gpii/node_modules/preferencesServer/src/preferencesServer.js

```diff
@@ -44,10 +44,15 @@ fluid.defaults("gpii.preferencesServer", {
             type: "gpii.preferencesServer.get.handler"
         },
         preferencesPost: {
-            route: "/preferences",
+            route: "/preferences/:gpiiKey",
```

Comment:
Why do we have preferencesPost in addition to preferencesPut?

Reply:
POST is to create a new gpiiKey and its vault. 

PUT is to update the vault of an exiting gpiiKey. If the key doesn't exist, an error will be returned.